build-tools/packages/build-cli/src/BasePackageCommand.ts

```diff
@@ -43,7 +43,7 @@ export abstract class PackageCommand<
 	 * @param pkg - The package being processed.
 	 * @param kind - The kind of the package.
 	 */
-	protected abstract processPackage(pkg: Package, kind: PackageKind): Promise<void>;
+	protected abstract processPackage<U extends Package>(pkg: U, kind: PackageKind): Promise<void>;
```

Comment:
Nit: Could / should this be something like `TPackage` instead of `U`? `U` doesn't convey a lot of info.

Might also be worth adding a `@typeParam` entry to the comment.